tests/test_aging.c

```diff
@@ -0,0 +1,532 @@
+        if (mkdir(dup, mode) == -1) {
+            if (errno != EEXIST) {
+                *p = '/';
+                free(dup);
+                return -1;
+            }
+        }
+        *p = '/';
+    }
+    free(dup);
+
+    if (mkdir(path, mode) == -1) {
+        if (errno != EEXIST) {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+static int
+create_file(char *path, char *content)
+{
+    FILE *fp;
+    fp = fopen(path, "w+");
+    chmod(path, 0700);
+    if (!fp) {
+        fprintf(stderr, "Creating a file %s failed\n", path);
+        return EXIT_FAILURE;
+    }
+    if (content) {
+        fprintf(fp, "%s", content);
+    }
+    fclose(fp);
+    return 0;
+}
+
+static int
+remove_file(void **state, char *file_name, int notif_dir)
+{
+    test_data_t *data = (test_data_t *) (*state);
+    if (asprintf(&data->path_to_file, "%s%s", (notif_dir) ? data->n_path : data->a_path, file_name) == -1) {
+        fprintf(stderr, "Asprintf failed\n");
+    }
+    if (remove(data->path_to_file)) {
+        fprintf(stderr, "Removing a file %s failed!\n", file_name);
+    }
+    free(data->path_to_file);
+    data->path_to_file = NULL;
+    return 0;
+}
+
+static int
+create_config(sr_session_ctx_t *sess, char *older_than, char *archive_dir, char *compress)
+{
+    int rc;
+    const struct ly_ctx *ctx;
+    struct lyd_node *node = NULL;
+
+    ctx = sr_session_acquire_context(sess);
+    if ((rc = lyd_new_path(NULL, ctx, "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/enabled/older-than",
+            older_than, 0, &node)) != LY_SUCCESS) {
+        return rc;
+    }
+    if ((rc = lyd_new_path(node, ctx, "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/enabled/archive-dir",
+            archive_dir, 0, NULL)) != LY_SUCCESS) {
+        return rc;
+    }
+    if ((rc = lyd_new_path(node, ctx, "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/enabled/compress",
+            compress, 0, NULL)) != LY_SUCCESS) {
+        return rc;
+    }
+    sr_session_release_context(sess);
+
+    if ((rc = sr_replace_config(sess, "sysrepo-plugind", node, 0)) != SR_ERR_OK) {
+        return rc;
+    }
+    return 0;
+}
+
+static int
+wait_for_archivation(void **state, uint64_t archived_expected)
+{
+    test_data_t *data = (test_data_t *) (*state);
+    sr_data_t *sr_data;
+    time_t start;
+    uint64_t num = 0;
+
+    sr_session_switch_ds(data->sess, SR_DS_OPERATIONAL);
+    time(&start);
+    while (time(NULL) < (start + 5)) {
+        if ((sr_get_data(data->sess, "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/archived-files-count",
+                0, 0, 0, &sr_data) != SR_ERR_OK)) {
+            return EXIT_FAILURE;
+        }
+        if (sr_data) {
+            num = strtoul(lyd_get_value(lyd_child(lyd_child(lyd_child(sr_data->tree)))), NULL, 10);
+            sr_release_data(sr_data);
+        }
+        if (num >= archived_expected) {
+            break;
+        }
+        usleep(TIME_MAX);
```

Comment:
`TIME_MAX` is not a suitable name for this macro.

Reply:
Renamed